code/modules/mob/living/living_defense.dm

```diff
@@ -257,24 +257,44 @@
 	spawn(1) updatehealth()
 	return 1
 
-/mob/living/proc/IgniteMob()
+/mob/living/proc/IgniteMob(var/fire_stacks_to_add = 0)
+
+	if(fire_stacks_to_add)
+		adjust_fire_stacks(fire_stacks_to_add)
+
 	if(fire_stacks > 0 && !on_fire)
 		on_fire = 1
 		set_light(light_range + MOB_FIRE_LIGHT_RANGE, light_power + MOB_FIRE_LIGHT_POWER)
 		update_fire()
+		return TRUE
+		
+	return FALSE
+
+/mob/living/proc/ExtinguishMob(var/fire_stacks_to_remove = 0)
 
-/mob/living/proc/ExtinguishMob()
-	if(on_fire)
+	if(fire_stacks_to_remove)
+		adjust_fire_stacks(-fire_stacks_to_remove)
+	else
+		fire_stacks = min(0,fire_stacks)

```

Comment:
Speaking of sane default behaviour. _This is not it. Absoloutely not._

Imagine a scenario where you automatically calculate how much you want to extinguish a mob. With this implementation, you'd have to guard against that amount:
* being negative
* being 0

Being able to insta-extinguish someone is fine, but it should be a different parameter that you have to pass explicitly. The way it is done now is prone to failure later on.

Reply:
Sure. I'll fix that.